hpat/tests/test_series.py

```diff
@@ -1037,13 +1037,60 @@ def test_impl(S):
         S = pd.Series(['aa', 'bb', np.nan])
         self.assertEqual(hpat_func(S), test_impl(S))
 
-    def test_series_mean1(self):
+    def test_series_mean(self):
         def test_impl(S):
             return S.mean()
         hpat_func = hpat.jit(test_impl)
 
-        S = pd.Series([np.nan, 2., 3.])
-        self.assertEqual(hpat_func(S), test_impl(S))
+        data_samples = [
+            [6, 6, 2, 1, 3, 3, 2, 1, 2],
+            [1.1, 0.3, 2.1, 1, 3, 0.3, 2.1, 1.1, 2.2],
+            [6, 6.1, 2.2, 1, 3, 3, 2.2, 1, 2],
+            [6, 6, np.nan, 2, np.nan, 1, 3, 3, np.inf, 2, 1, 2, np.inf],
+            [1.1, 0.3, np.nan, 1.0, np.inf, 0.3, 2.1, np.nan, 2.2, np.inf],
+            [1.1, 0.3, np.nan, 1, np.inf, 0, 1.1, np.nan, 2.2, np.inf, 2, 2],
+            [np.nan, np.nan, np.nan],
+            [np.nan, np.nan, np.inf],
+        ]
+
+        for data in data_samples:
+            S = pd.Series(data)
+            with self.subTest(S=S):
```

Comment:
```suggestion
            with self.subTest(S=data):
```
I'm not sure it will be good to read output of the test if variations based on `S`